clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -42,8 +42,22 @@ public KafkaChannel(String id, TransportLayer transportLayer, Authenticator auth
     }
 
     public void close() throws IOException {
-        transportLayer.close();
-        authenticator.close();
+        IOException exception = null;
+        try {
+            transportLayer.close();
+        } catch (IOException e) {
+            exception = e;
+        }
+        try {
+            authenticator.close();
+        } catch (IOException e) {
+            if (exception != null)
+                exception.addSuppressed(e);
+            else
+                exception = e;
+        }
+        if (exception != null)
+            throw exception;
```

Comment:
Ouch, this is more awkward than I thought. How about we create a method in `Utils` that takes a varargs of `Closeables` and does something along these lines for the whole collection?